crates/driver/src/domain/competition/mod.rs

```diff
@@ -403,9 +403,11 @@ impl Competition {
                 let settle_fut =
                     self.process_settle_request(auction_id, solution_id, submission_deadline);
                 let result = tokio::select! {
-                    // Check whether the sender is closed, which indicates that the client
-                    // has disconnected and the settlement task needs to be cancelled.
+                    // Cancel the settlement task if the sender is closed (client likely disconnected).
+                    // This is a fallback to recover from issues like a stuck driver (e.g., stalled block stream).
                     _ = response_sender.closed() => {
+                        // Add a grace period to give driver the last chance to fetch the settlement tx.
+                        tokio::time::sleep(Duration::from_secs(1)).await;
```

Comment:
The grace period only helps if we continue polling the `settle_fut` in the meantime.